src/AutoRest.CSharp/Common/Output/Builders/SerializationBuilder.cs

```diff
@@ -425,6 +425,8 @@ public JsonObjectSerialization BuildJsonObjectSerialization(InputModelType input
             PopulatePropertyBag(propertyBag, 0);
             PopulatePropertyBag(selfPropertyBag, 0);
 
+            // properties: all the properties containing the properties from base needed to build the constructor
+            // selfProperties: the properties not containing properties from base just used to do the serialization
             var properties = GetPropertySerializationsFromBag(propertyBag, objectType).ToArray();
```

Comment:
I am more curious why we need both.
If this is a temporary solution, please add TODO and/or open an issue to track.
If this is permanent, I think we need comments to explain why we need both.
The serialization objects might not have the hierarchy and we have to do this, this is an acceptable explanation, but we need to write them somewhere so that in the future we will not be confused.